deployment/ec2-with-asg.tf

```diff
@@ -0,0 +1,162 @@
+# IAM Role and Policy for EC2 Instances
+resource "aws_iam_role" "ec2_role" {
+  name_prefix = "${local.environment_prefix}ec2_role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Action = "sts:AssumeRole",
+        Effect = "Allow",
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        },
+      },
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "ec2_elb_access" {
+  role       = aws_iam_role.ec2_role.name
+  policy_arn = "arn:aws:iam::aws:policy/ElasticLoadBalancingReadOnly"
+}
+
+resource "aws_iam_role_policy_attachment" "ec2_describe_ec2" {
+  role       = aws_iam_role.ec2_role.name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ReadOnlyAccess"
+}
+
+resource "aws_iam_role_policy_attachment" "ec2_cloudwatch_logs" {
+  role       = aws_iam_role.ec2_role.name
+  policy_arn = "arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy"
+}
+
+resource "aws_iam_instance_profile" "ec2_profile" {
+  name_prefix = "${local.environment_prefix}ec2_profile"
+  role        = aws_iam_role.ec2_role.name
+}
+
+# ASG with launch template
+resource "aws_launch_template" "ec2_launch_templ" {
+  name_prefix = "${local.environment_prefix}ec2_launch_templ"
+  image_id    = "ami-05e00961530ae1b55"
+  # instance_type = "t2.micro"
+  instance_type = "t2.large"
+  user_data = base64encode(templatefile("user_data.sh.tpl", {
+    LOG_FILE              = "/var/log/user_data.log"
+    BRANCH_NAME_TO_DEPLOY = data.dotenv.env_file.env["BRANCH_NAME_TO_DEPLOY"]
+    TARGET_GROUP_NAME     = aws_lb_target_group.alb_tg.name
+    environment_prefix    = local.environment_prefix
+    DATABASE_URL          = data.dotenv.env_file.env["DATABASE_URL"]
+    SECRET_KEY_BASE       = data.dotenv.env_file.env["SECRET_KEY_BASE"]
+    POOL_SIZE             = data.dotenv.env_file.env["POOL_SIZE"]
+    BEARER_TOKEN          = data.dotenv.env_file.env["BEARER_TOKEN"]
+    PORT                  = data.dotenv.env_file.env["PORT"]
```

Comment:
Also include a .env.example in the PR with all these variables?

Reply:
Added